performance/front/table-and-chart-with-good-performance.md

```diff
@@ -0,0 +1,41 @@
+# A table or a chart on my project takes to much time to render
+
+## Owner: [Justine Mignot](https://github.com/justinemignot)
+
+## Do you need to display all data you are displaying currently ?
```

Comment:
So you have two parts, maybe add steps directly after the main title saying : 
```markdown
## Steps

- [ ] Find how many lines or data points are displayed on your page
- [ ] Decide if your end user should see them all
  - [ ] If it does not need all this data, try reducing its amount
  - [ ] If it does need all the data, see how to optimize render
  - Find solutions per components, "Charts, Maps, Tables and Lists" below 
```

Reply:
And then I would have sub-parts under "## Steps" with
- `### Chart/Maps data optimization` **-> here develop the idea of displaying less**
- `### Tables/Lists render optimization` **-> here talk about fetching less and/or displaying less, virtual render, etc.**

Reply:
And under each of those sub-topics, I would first have steps on what to do, like :
```markdown
- Open your page with the chart
- Reduce by half the quantity of data displayed
- If the result is still clear enough for the end user, check how long it took to render (link to Thibaut's article on how to investigate)
  - If the improvement is enough and the information clear for the user, perfect, you improved your performance and still helped your end user
  - If it's not enough, reproduce the previous step
  - If you can't manage to improve the performance and/or that you are now displaying too few points, you should investigate the render performance of your components, it means that the code is too heavy, not that you display too much data
```
And you can put the story of why it is so important above with the ">", like :
```markdown
> On a project like Investo for BNP, we reduced by X% the data points of a chart, the page render time went from Ys to Zs
```